pincer/commands/arg_types.py

```diff
@@ -0,0 +1,228 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+from typing import Any, List, Tuple, Union, T
+
+from ..utils.types import MISSING
+from ..objects.app.command import AppCommandOptionChoice
+
+
+class _CommandTypeMeta(type):
+    def __getitem__(cls, args: Union[Tuple, Any]):
+        if not isinstance(args, tuple):
+            args = args,
+
+        return cls(*args)
+
+
+class CommandArg(metaclass=_CommandTypeMeta):
+    """
+    Holds all application command options
+
+    .. code-block:: python3
+
+        CommandArg[
+            # This is the type of command.
+            # Supported types are str, int, bool, float, User, Channel, and Role
+            int,
+            # The modifiers to the command go here
+            Description["Pick a number 1-10"],
+            MinValue[1],
+            MaxValue[10]
+        ]
+
+    Parameters
+    ----------
+    command_type : T
+        The type of the command
+    *args : :class:`pincer.commands.arg_types.Modifier`
+
+    """
+
+    def __init__(self, command_type, *args) -> None:
+        self.command_type = command_type
+        self.modifiers = args
+
+    def get_arg(self, arg_type: T) -> T:
+        for arg in self.modifiers:
+            if isinstance(arg, arg_type):
+                return arg.get_payload()
+
+        return MISSING
+
+
+class Modifier(metaclass=_CommandTypeMeta):
+    """
+    Modifies a CommandArg by being added to
+    :class:`pincer.commands.arg_types.CommandArg`'s args.
+    """
+
+
+class Description(Modifier):
+    """
+    Represents the description application command option type
```

Comment:
```suggestion
    Represents A description of a Application Command
```

Reply:
@RPSMain you didn't really improve it tbh
```suggestion
    Represents the description of an application command option type
```